### ISSUE TITLE
Tracking BatchNorm FW/BW support

### ISSUE THREAD

**Issue by jjsjann123** [csarofeen/pytorch#718]
Tracking work in progress to support BN fw/bw:

My working branch is: https://github.com/csarofeen/pytorch/tree/rds_batch_norm_stolen
This has merged Christian's fixes for codegen from norm_hack_2_rebased(?)

TODO list:

- [x] Support output composition with inplace aliases in segmented fusion;
- [x] Update BN in parser to use composite BN op with different backend;
- [ ] Allow output of const scalar in fusion; (no longer needed)
- [x] Track bug status: #715 #716 #717;
- [ ] multiple dimensional BN; (This should be an op parsing thing that can be tackled when it show up)
- [x] open issue on trivial reduction encountered in python tests;
- [x] python tests with functional BN forward/backward codegen kernel;
- [x] python tests for torch.nn.BatchNorm API where tracking stats & affine transforms are optional. (as well as dynamic eps & momentum)

**Comment by csarofeen** [comment]
@jjsjann123 is there anything left to do?

**Comment by jjsjann123** [comment]
Closing as items are either worked out or irrelevant.
